### PR TITLE
[WC-2474]: Handle keyboard nav for input elements in Gallery

### PR DESCRIPTION
**Pull request by rahmanunver**
### Pull request type

Bug fix (non-breaking change which fixes an issue)

---

### Description
Added a new handler method to check if the element is an HTMLInputElement, if so stop the propagation.

### What should be covered while testing?
Demo project link and #frontend thread:
https://mendix.slack.com/archives/CJZ85RLTA/p1714654971335249

Input elements inside Gallery widget should now be able to use arrow keys to move cursor when focused, but this means where are stopping event propagation for the input elements, which means keyboard navigation behavior for text inputs are going to change.

### CODE REVIEW

File: packages/pluggableWidgets/gallery-web/src/components/ListItem.tsx [hunk 1]
```diff
@@ -26,6 +26,17 @@ export function ListItem(props: ListItemProps): ReactElement {
...
+        // Handling the case of element being an input element, to be able to move cursor with keyboard
+        if (
+            event.target instanceof HTMLInputElement &&
+            ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"].includes(event.key)
+        ) {
+            event.stopPropagation();
+        } else {
+            handlers?.onKeyDown?.(event);
+        }
+    };
```
Comment (iobuhov) [inline packages/pluggableWidgets/gallery-web/src/components/ListItem.tsx:39]: Can you also please move this logic to events controller?

File: packages/pluggableWidgets/gallery-web/src/components/ListItem.tsx [hunk 2]
```diff
@@ -26,6 +26,17 @@ export function ListItem(props: ListItemProps): ReactElement {
     const keyNavProps = useFocusTargetProps({ columnIndex: columnIndex ?? -1, rowIndex });
     const handlers = useMemo(() => eventsController.getProps(item), [eventsController, item]);
 
+    const handleOnKeyDown = (event: React.KeyboardEvent<HTMLDivElement>): void => {
+        // Handling the case of element being an input element, to be able to move cursor with keyboard
+        if (
+            event.target instanceof HTMLInputElement &&
+            ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"].includes(event.key)
```
[the remark this answers is not in the document]
Comment (rahmanunver) [reply packages/pluggableWidgets/gallery-web/src/components/ListItem.tsx]: Updated

File: packages/pluggableWidgets/gallery-web/src/components/ListItem.tsx (hunk 1 above)
Comment (rahmanunver) [reply packages/pluggableWidgets/gallery-web/src/components/ListItem.tsx]: Updated

**Review by iobuhov — DISMISSED** [review]
LGTM

**Review by samuelreichert — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Review by iobuhov — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**